plugins/chat/app/models/chat/message.rb

```diff
@@ -91,11 +95,57 @@ class Message < ActiveRecord::Base
 
     before_save { ensure_last_editor_id }
 
+    normalizes :blocks,
+               with: ->(blocks) do
+                 return if !blocks
+
+                 # automatically assigns unique IDs
+                 blocks.each do |block|
+                   block["block_id"] = SecureRandom.uuid if !block["block_id"]
+                   block["elements"].each do |element|
+                     element["action_id"] = SecureRandom.uuid if element["type"] == "button" &&
+                       !element["action_id"]
```

Comment:
Could you do this instead?

```suggestion
                     element["action_id"] ||= SecureRandom.uuid if element["type"] == "button"
```